src/core/ddsi/src/q_receive.c

```diff
@@ -2896,7 +2896,7 @@ static bool do_packet (struct thread_state1 * const ts1, struct q_globals *gv, d
 
   if (conn->m_stream)
   {
-    MsgLen_t * ml = (MsgLen_t*) (buff + RTPS_MESSAGE_HEADER_SIZE);
+    MsgLen_t * ml = (MsgLen_t*) (hdr + 1);
```

Comment:
Why exactly is this change required? The address is the same, that's not the issue.

Secondly, there is actually a real bug a few lines below, which is that it doesn't verify the received message length is within bounds ...

Reply:
The address is the same indeed, but the code is in my opinion easier to read (if hdr == buff then buff + sizeof(header_t) == hdr + 1.
I don't see the real bug. As far as I can see all the received messages are within bounds.

Reply:
The problem I alluded to is in this line:
```C
        sz = ddsi_conn_read (conn, buff + stream_hdr_size, ml->length - stream_hdr_size, false, NULL);
```

and is simply that there is no check that ``ml->length`` is ≥``stream_hdr_size``, nor that it is ≤ than ``maxsz``.